context_header.go

```diff
@@ -36,31 +36,56 @@ type ContextWithHeaders interface {
 	SetResponseHeaders(map[string]string)
 }
 
+type headersContextKey int
```

Comment:
we should be able to reuse `contextKey`
https://github.com/uber/tchannel-go/blob/master/context.go#L31